File: 8th-May-Binary_Modulo/c#/solution.cs
/*
This Code is written by Bhaskar
*/

public class Solution
{
    public int Modulo(string s, int m)
    {
        int k = 0;

        // Convert the binary string to an integer
        for (int i = 0; i < s.Length; i++)
        {
            k = ((k << 1) % m + (s[i] - '0') % m) % m;
        }

        return k;
    }
}

File: 8th-May-Binary_Modulo/python/solution.py

#This Code is written by Bhaskar


class Solution:
    def modulo(self, s, m):
        k = 0

        # Convert the binary string to an integer
        for i in range(len(s)):
            k = ((k << 1) % m + (ord(s[i]) - ord('0')) % m) % m

        return k

